api/server/httputils/errors.go

```diff
@@ -54,6 +54,7 @@ func GetHTTPErrorStatusCode(err error) int {
 			code    int
 		}{
 			{"not found", http.StatusNotFound},
+			{"cannot find", http.StatusNotFound},
```

Comment:
Can we avoid expanding this "guess the status code" system further and return a specific http error instead?